core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -320,6 +320,12 @@ private[spark] class ExecutorAllocationManager(
     }
   }
 
+  // SPARK-47474: Please do not delete this function, the tests in `ExecutorAllocationManagerSuite`
+  // need to access `listener.totalRunningTasksPerResourceProfile` with `synchronized`.
+  private def totalRunningTasksPerResourceProfile(id: Int): Int = synchronized {
```

Comment:
cc @dongjoon-hyun  @mridulm @tgravescs

Since new comments were added during the revert, I have created a new Jira. Is there a better way to describe the content of the comments?